contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/DrtTripsAnalyser.java

```diff
@@ -472,4 +477,17 @@ public static String summarizeDetailedOccupancyStats(
 
 		return result.toString();
 	}
+	
+	public static double getSatisfactionRate(DescriptiveStatistics stats, int timeCriteria) {
+		double[] sortedWaitingTime = stats.getSortedValues();
+		for (int i = 0; i < sortedWaitingTime.length; i++) {
+			if (sortedWaitingTime[i] > timeCriteria) {
+				double numOfRequestsInZone = sortedWaitingTime.length;
+				double output = i / numOfRequestsInZone;
+				return output;
+			}
+		}
```

Comment:
since data is sorted, we could use a binary search https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Arrays.html#binarySearch(double%5B%5D,double)

Reply:
👍 I will try with that. Thanks!

Reply:
Oh, I just looked into `DescriptiveStatistics.getSortedValues()`. It performs sorting every time it is called. So my final recommendation is to:
- call `stats.getValues()`
- loop though the whole array and count the positives